spec/defra_ruby/exporters/registration_bulk_export_report_spec.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe DefraRuby::Exporters::RegistrationBulkExportReport do
+  describe "#COLUMNS" do
+    it "includes all of the headings required to generate the export" do
+      actual_headers = described_class::COLUMNS.map { |h| h[:header] }
+      expected_headers = %w[
+        reference_number
+        registration_date
+        applicant_first_name
+        applicant_last_name
+        applicant_phone_number
+        applicant_email_address
+        organisation_type
+        company_reference_number
+        organisation_name
+        organisation_premises
+        organisation_street_address
+        organisation_locality
+        organisation_city
+        organisation_postcode
+        correspondance_contact_first_name
+        correspondance_contact_last_name
+        correspondance_contact_position
+        correspondance_contact_email_address
+        correspondance_contact_phone_number
+        correspondance_contact_premises
+        correspondance_contact_street_address
+        correspondance_contact_locality
+        correspondance_contact_city
+        correspondance_contact_postcode
+        on_a_farm?
+        is_a_farmer?
+        site_premises
+        site_street_address
+        site_locality
+        site_city
+        site_postcode
+        site_country
+        site_location_grid_reference
+        site_location_description
+        exemption_code
+        exemption_summary
+        exemption_status
+        exemption_valid_from_date
+        exemption_expiry_date
+        exemption_deregister_date
+        exemption_deregister_comment
+        assistance_type
+      ]
+      expect(actual_headers).to match_array(expected_headers)
+    end
+  end
+
+  let(:registrations) do
+    [
+      create(:registration, exemptions: WasteExemptionsEngine::Exemption.first(7)),
+      create(:registration, exemptions: WasteExemptionsEngine::Exemption.first(3)),
+      create(:registration, exemptions: WasteExemptionsEngine::Exemption.first(5))
+    ]
+  end
+
+  it "returns records ordered by their registration date ascending" do
+    Helpers::Registrations.activate(registrations[0], Date.new(2019, 3, 3))
+    Helpers::Registrations.activate(registrations[1], Date.new(2019, 3, 11))
+    Helpers::Registrations.activate(registrations[2], Date.new(2019, 3, 7))
+
+    query_reg_ids = described_class.query.each.map { |re| re.registration.id }
+    expect(query_reg_ids.uniq).to eq([registrations[0].id, registrations[2].id, registrations[1].id])
+  end
+
+  describe "#query" do
+    context "when there are multiple registrations with multiple exemptions" do
+      context "and all are active" do
+        before(:each) do
+          registrations.each do |registration|
+            Helpers::Registrations.activate(registration)
+          end
+        end
+
+        it "returns the correct number of records" do
```

Comment:
The `it` message is right and consistent with the EPR tests. However I think we can make the expected behaviour clearer if we said something like

```ruby
it "returns all records" do
```

Reply:
Fixed in ddd1354.